### PR TITLE
gh-116326: Handler errors correctly in `getwindowsversion` in `sysmodule`

### PR DESCRIPTION
**Pull request by sobolevn**
- `PyErr_Occurred` is no longer needed, because we check for all `NULL` values
- I've also added an explicit error check when `Py_BuildValue("(kkk)", ...)` can fail

* Issue: gh-116326

### CODE REVIEW

**Review by colesbury — APPROVED** [review]
LGTM!

**Comment by sobolevn** [comment]
Thanks! I would like to wait for a bit for @ericsnowcurrently's review as a code owner :)

File: Python/sysmodule.c
Comment (serhiy-storchaka) [inline Python/sysmodule.c]: It is no longer needed.

**Review by serhiy-storchaka — COMMENTED** [review]
`PyErr_Clear()` calls in this function looks suspicious too. In good code it should only be called for expected exceptions, the others should be reported as errors.

File: Python/sysmodule.c
Comment (sobolevn) [reply Python/sysmodule.c]: I think it is needed, because `realVersion` is set to 
```c
        Py_BuildValue("(kkk)",
            ver.dwMajorVersion,
            ver.dwMinorVersion,
            ver.dwBuildNumber
        );
```

which can be `NULL`.
So, if it is `NULL` we go to `error`.

**Comment by sobolevn** [comment]
> In good code it should only be called for expected exceptions, the others should be reported as errors.

I think that this is a side-effect of using `PyObject_GetAttrString`, I can change it to be `PyObject_GetOptionalAttrString` 👍

File: Python/sysmodule.c
Comment (serhiy-storchaka) [reply Python/sysmodule.c]: But SET_VERSION_INFO below checks for NULL.

**Comment by serhiy-storchaka** [comment]
Yes, one of `PyErr_Clear`s is for AttributeError, and `PyObject_GetOptionalAttrString` is a good alternative. There is another one, the exception type should be checked before it.

File: Python/sysmodule.c
Comment (sobolevn) [reply Python/sysmodule.c]: Indeed 👍 
Thank you!

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Comment by sobolevn** [comment]
@serhiy-storchaka what is the best way to do backports here? I forgot that `PyObject_GetOptionalAttrString` is 3.13+ only.

**Comment by serhiy-storchaka** [comment]
Explicitly check for AttributeError.

**Comment by sobolevn** [comment]
This failure also happened on a previous PR: https://buildbot.python.org/all/#/builders/1102/builds/751/steps/5/logs/stdio

Plus, it is a debian failure for a windows-only change.

**Pull request merged**
